From 8f2f6c3fb184c6a5ade34a74c5b58f96cd27cf19 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Fri, 31 Oct 2014 17:45:04 +0100
Subject: [PATCH 074/117] fish: complete symlink properly

When an entry is a symlink, check whether it points to a directory, so
the entry for it can be completed correctly.

(cherry picked from commit 7ba102dc4cd4b7c80f7486a54bf8535dd964f436)
---
 fish/destpaths.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fish/destpaths.c b/fish/destpaths.c
index 20ac56f..fbf49e3 100644
--- a/fish/destpaths.c
+++ b/fish/destpaths.c
@@ -191,6 +191,7 @@ complete_dest_paths_generator (const char *text, int state)
                     words = w;
                     words[nr_words].name = p;
                     if (dirents->val[i].ftyp == 'u'
+                        || dirents->val[i].ftyp == 'l'
                         || dirents->val[i].ftyp == '?') {
                       CLEANUP_FREE char *rp = guestfs_realpath (g, words[nr_words].name);
                       if (rp)
-- 
2.5.0

